### ISSUE TITLE
Label Paper S10: Variational Bayes Block Sparse Modeling with Correlated Entries

### ISSUE THREAD

**Issue by kerautret** [RLPR/LabelReviews#10]
# General info
- [x] Conference: ICPR 2018
- [x] Papers title: Variational Bayes Block Sparse Modeling with Correlated Entries
- [x] Source paper:  [paper SXX](https://github.com/kerautret/PrivatePDFforRLPR/blob/master/sXX.pdf)
- [x] GitHub repo: 
https://github.com/shruti51/Variational-Bayes-Block-Sparse-Modeling-with-Correlated-Entries
- [x] Submission date: 5/08/2018
- [x] Platform: Window 10
- [x] Language: Matlab 
- [x] Results to be reproduced: (indicates reference to figure/tab) 
   -  fig2_and_fig3.m

- [ ] Reproduction main step instructions: from the [Readme](https://github.com/shruti51/Variational-Bayes-Block-Sparse-Modeling-with-Correlated-Entries/blob/master/README.md) file:
      ` the directory, the demo codes titled as demo_figX.m generate the X figure number in the paper. Demo codes titled as demo_figY_and_figZ.m generate the X and Y figure number in the paper as per the instructions given in the comment section of the corresponding code.`

# Reviewer feedback
 - [ ] Tested platform: 
 - [ ]  Result reproduced: 
 - [ ]  Reviewer comment:

## Details Results
- Execution without errors:
    - [ ] without changes
    - [ ] with small changes
    - [ ] with a lot of changes
- Results are obtained:
    - [ ] always
    - [ ] sometimes
    - [ ] never
- Results correspond to the paper's figures/tables:
    - [ ] exactly
    - [ ] with negligible differences
    - [ ] with differences
    - [ ] with a lot of differences

**Comment by akrah** [comment]
Dear @shruti51, I started the review and I have the following requests:
1) Actually, your code take a long time to be executed and required to change parameters in the code for each sub-figure. Could you please modify your scripts `demo_fig2_and_fig3.m` and `demo_fig4.m` with loops on the `status` and `rho` parameters, and replace the figure drawing by a saving in order to generate all sub-images without human intervention
2) Moreover, could you separate figures 2 and 3? I know that this will duplicate code but it will be more reproducible to do each figure in a separated script.

Thanks!

**Comment by akrah** [comment]
@shruti51, I have an error and a warning at the end of execution of `demo_fig4.m`.
The warning message:
```
Warning: Ignoring extra legend entries. 
> In legend>set_children_and_strings (line 646)
  In legend>make_legend (line 316)
  In legend (line 259)
  In demo_fig4 (line 251)
```
The error message:
```
Index exceeds matrix dimensions.

Error in demo_fig4 (line 271)
plot(0.1:0.05:0.5,(1-(succ(:,7)/k)),'-.k*')
```
I suppose (but I'm not sure) that this is due to the lines 218-219, because now `succ` is smaller than the commented line:
```
% succ=[succ; [succ_BSBL_EM succ_BSBL_BO succ_JTSVB succ_JSVB succ_LTSVB succ_LSVB succ_StTSVB succ_StSVB]]
  succ=[succ; [succ_JTSVB succ_JSVB succ_LTSVB succ_LSVB succ_StTSVB succ_StSVB]]
```
Please could you fix that (and take into account the previous comment)?

Thanks!

**Comment by shruti51** [comment]
Thank you for your comments. I have made all the necessary changes and uploaded updated files. 

P.S. As far as speed is concerned, actually for each block size and rho, 6 different algorithms are running for 1000 independent trials which are making the code time-consuming. To combat this issue, I have made changes where instead of 1000 trials algorithms will run only for 100 trials. Though results will not be exactly the same but will reflect what is there in the paper. To reproduce exact results, please change the following loop in demo_fig2, demo_fig3 and demo_fig4:
for k=1: 100
---
---
---
end

**Comment by akrah** [comment]
Hi @shruti51, thanks for your changes!
Could you just:
    - apply the last point of the first request: change the figure displaying into a figure saving? This could be useful to generate all figures without human intervention
    - as for fig 2 and 3, make the loops on rho and take this parameter into account in the figure caption?
Thanks!

**Comment by shruti51** [comment]
Okay. Do you want me to save the figures to say, .png file? For example, should I save as 'Fig2_rho=0.png' for the first figure of Fig.2?

**Comment by akrah** [comment]
Yes, in PNG or as Matlab figure or both, as you prefer whith a naming like you propose.

**Comment by shruti51** [comment]
I have updated the code of demo_fig.2, demo_fig.3 and demo_fig.4  as per your suggestions.

**Comment by akrah** [comment]
Thanks! It works but for script `demo_fig4.m`, the same figure is saved two times for RelativeReconstructionError and FailureRate.  Could you fix this?

**Comment by shruti51** [comment]
I apologize for that minute bug. I have fixed it now.
